microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,293 @@
+
+**Detailed contents:**
+
+* [Events for processes that exited or failed](#events-for-processes-that-exited-or-failed)
+* [Gathering process failure details](#gathering-process-failure-details)
+* [Handle process crashes](#handle-process-crashes)
+   * [The main browser process has exited unexpectedly](#the-main-browser-process-has-exited-unexpectedly)
+   * [A process rendering content in the WebView2 control has exited unexpectedly](#a-process-rendering-content-in-the-webview2-control-has-exited-unexpectedly)
+   * [The GPU process has exited unexpectedly](#the-gpu-process-has-exited-unexpectedly)
+   * [A utility process has exited unexpectedly](#a-utility-process-has-exited-unexpectedly)
+   * [Any other process has exited unexpectedly](#any-other-process-has-exited-unexpectedly)
+* [Handle unresponsive renderers](#handle-unresponsive-renderers)
+* [Handle main browser process exited](#handle-main-browser-process-exited)
+   * [Clearing the user data folder](#clearing-the-user-data-folder)
+   * [Updating the WebView2 Runtime](#updating-the-webview2-runtime)
+   * [Restarting with a different environment configuration](#restarting-with-a-different-environment-configuration)
+   * [Clearing the auth cache](#clearing-the-auth-cache)
+* [See also](#see-also)
+
+
+<!-- ====================================================================== -->
+## Events for processes that exited or failed
+
+When you initialize a WebView2 control, WebView2 will ensure there's a WebView2 Runtime to power your control and connect to its [WebView2 Process Group](process-model.md#processes-in-the-webview2-runtime). Once this connection is established, your control will start monitoring these processes and report the following events so your application can react accordingly:
+
+* **Any process failure.**  When _any of the processes_ in the WebView2 Runtime fails, the CoreWebView2 will raise the `ProcessFailed` event. This can be due to a process crash, or an unresponsive renderer process. Use this event for diagnostics and recovery from failures in the WebView2 processes.  See [Handle process crashes](#handle-process-crashes) and [A process rendering content in the WebView2 control has exited unexpectedly](#a-process-rendering-content-in-the-webview2-control-has-exited-unexpectedly), below.
+
+* **Main browser process exits.**  If the main browser process exits for _any reason_, the `CoreWebView2Environment` will raise the `BrowserProcessExited` event. Use this event to synchronize operations involving the WebView2 Runtime resources and lifetime, such as _User Data Folder_ management and updates. See [Handle main browser process exited](#handle-main-browser-process-exited), below.
+
+* **Main browser process crashes.**  When the main browser process crashes, it will produce both a `ProcessFailed` event and a `BrowserProcessExited` event, since the main browser process _exited_ because of a failure.  <!--todo: link to a section below?-->
```

Comment:
@mikehoffms this is more like a note, and is covered in the detailed sections for the two cases above. I don't think we need links here.

Reply:
locally, removed comment